### PR TITLE
fix: added errors to internal implementation of ReadStorage

### PR DESCRIPTION
**Pull request by vbar**
# What :computer: 
Make functions implementing `ReadStorage` for `ForkStorage` return `Result::Err` on failure, and call them in preference to the panicking interface.

# Why :hand:
Showing a network error is preferable to server panic.

# Evidence :camera:
Both unit and e2e tests pass.

# Notes :memo:
Unfortunately the `ReadStorage` trait cannot be removed, because it's used by zkSync Era - for a clean solution, it should be updated (so that its methods return `Result`) there.

### CODE REVIEW

File: src/fork.rs [hunk 1]
```diff
@@ -182,15 +181,15 @@ impl<S: ForkSource> ForkStorage<S> {
 
 impl<S: std::fmt::Debug + ForkSource> ReadStorage for ForkStorage<S> {
     fn is_write_initial(&mut self, key: &StorageKey) -> bool {
-        self.is_write_initial_internal(key)
+        self.is_write_initial_internal(key).unwrap()
```
Comment (MexicanAce) [inline src/fork.rs:184]: Should these methods more gracefully handle `None`? We've tried to get rid/avoid `.unwrap()` as much as possible.

Comment (vbar) [reply src/fork.rs:184]: Well, they should, but how? They're implementing a trait which doesn't consider the possibility of failure, and which must be implemented to use `ForkStorage` inside EraVM...

Comment (MexicanAce) [reply src/fork.rs:185]: Discussed offline and agreed this is best to match existing EraVM logic/expectations

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**